lib/pure/os.nim

```diff
@@ -1997,6 +1997,8 @@ proc getFileInfo*(handle: FileHandle): FileInfo =
     rawToFormalFileInfo(rawInfo, result)
 
 proc getFileInfo*(file: File): FileInfo =
+  if file.isNil:
```

Comment:
While this helps with debugging does it mean that we should be checking every function param to see whether it is nil?